File: core/src/main/scala/stryker4s/run/report/html/mutantStatus.scala
package stryker4s.run.report.html

sealed trait MutantStatus

case object Killed extends MutantStatus
case object Survived extends MutantStatus
case object NoCoverage extends MutantStatus
case object CompileError extends MutantStatus
case object RuntimeError extends MutantStatus
case object Timeout extends MutantStatus

File: stryker4jvm-core/src/main/java/stryker4jvm/core/model/elements/MutantStatus.java
package stryker4jvm.core.model.elements;

public enum MutantStatus {
  /** Indicates that the mutant was killed. */
  Killed,
  /** Indicates that the mutant has survived. */
  Survived,
  /** Indicates that the mutant has not been covered at runtime. */
  NoCoverage,
  /** Indicates that the mutant caused a timeout during runtime. */
  Timeout,
  /** Indicates that the mutant could not be compiled at all. */
  CompileError,
  /** Indicates that the mutant caused a runtime exception. */
  RuntimeError,
  /** Indicates that the mutant was ignored entirely, most likely due to user configuration. */
  Ignored
}
